cmd/collectors/storagegrid/rest/client.go

```diff
@@ -380,27 +385,28 @@ func (c *Client) fetchTokenWithAuthRetry() error {
 				if err2 != nil {
 					return err2
 				}
-				return c.fetchToken()
+				return ci.fetchToken()
 			}
 		}
 	}
 	return err
 }
 
-func (c *Client) fetchToken() error {
+// fetchToken should not be used directly. Use fetchTokenWithAuthRetry instead.
+func (ci *clientInternal) fetchToken() error {
 	var (
 		err      error
 		req      *http.Request
 		response *http.Response
 		body     []byte
 	)
-	u, err := url.JoinPath(c.baseURL, c.APIPath, "authorize")
+	u, err := url.JoinPath(ci.client.baseURL, ci.client.APIPath, "authorize")
```

Comment:
may be get client first as below and then use same for all places?
`c := ci.client`